webbpsf/webbpsf_core.py

```diff
@@ -920,6 +921,87 @@ def load_was_opd(self, inputWasOpd, size=1024, save=False, filename='new_was_opd
 
         return fits.HDUList(fits.ImageHDU(newopd, header=hdr))
 
+    def psf_grid(self, all_detectors=True, num_psfs=16, single_psf_centered=True, use_detsampled_psf=False,
+                 save=False, outfile=None, overwrite=True, **kwargs):
+        """
+        Create a PSF library in the form of a grid of PSFs across the detector based on the specified instrument,
+        filter, and detector. The output file will contain a 3D array with axes [i, y, x] where i is the PSF position
+        on the detector grid and (y,x) is the 2D PSF.
+
+        Parameters
+        ----------
+        all_detectors : bool
+            If True, run all detectors for the instrument. If False, run for the detector set in
+            the instance. Default is True
+        use_detsampled_psf : bool
+            If True, the grid of PSFs returned will be detector sampled (made by binning down the
+            oversampled PSF). If False, the PSFs will be oversampled by the factor defined by the
+            oversample/detector_oversample/fft_oversample keywords. Default is False.
+        num_psfs : int
+            The total number of fiducial PSFs to be created and saved in the files. This
+            number must be a square number. Default is 16.
+            E.g. num_psfs = 16 will create a 4x4 grid of fiducial PSFs.
+        single_psf_centered : bool
+            If num_psfs is set to 1, this defines where that psf is located. If True it will be the
+            center of the detector, if False it will be the location defined in the WebbPSF
+            attribute detector_position (reminder - detector_position is (x,y)). Default is True
+        save : bool
+            True/False boolean if you want to save your file. Default is False.
+        outfile : str
```

Comment:
Instead of `save` + `outfile` I'd suggest a single keyword where `None` means "don't save", and otherwise the user has to give the filename.  If you still want the default behavior, you could use standard python string formatting and tell the user to do `'instr_det_filt_fovp{}_samp{}_npsf{}.fits'` or similar.
 
(Caveat: the comment I'm about to leave in the "general" thread might make both of these go away.)